django/contrib/gis/geos/prototypes/io.py

```diff
@@ -169,8 +169,11 @@ def read(self, wkb):
         if isinstance(wkb, memoryview):
             wkb_s = bytes(wkb)
             return wkb_reader_read(self.ptr, wkb_s, len(wkb_s))
-        elif isinstance(wkb, (bytes, str)):
+        elif isinstance(wkb, bytes):
             return wkb_reader_read_hex(self.ptr, wkb, len(wkb))
+        elif isinstance(wkb, str):
+            wkb_s = wkb.encode()
+            return wkb_reader_read_hex(self.ptr, wkb_s, len(wkb))
```

Comment:
I would have written `len(wkb_s)` as last argument of `wkb_reader_read_hex`

Reply:
I have made that change.